lib/pages/leaderboard_page/components/leaderboard_page_builder.dart

```diff
@@ -0,0 +1,71 @@
+import 'package:dalal_street_client/blocs/leaderboard/leaderboard_cubit.dart';
+import 'package:dalal_street_client/pages/leaderboard_page/components/leaderboard_page_data.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class LeaderboardPageBuilder extends StatefulWidget {
+  const LeaderboardPageBuilder({Key? key, required this.leaderboardType})
+      : super(key: key);
+
+  final String leaderboardType;
```

Comment:
```suggestion
  final LeaderboardType leaderboardType;
```
When accepting leaderboardType, declare it as enum and not as string.
Do this everywhere